src/frontend/deployment_details.go

```diff
@@ -3,27 +3,37 @@ package main
 import (
 	"net/http"
 	"os"
-	"sync"
+	"time"
 
 	"cloud.google.com/go/compute/metadata"
 	"github.com/sirupsen/logrus"
 )
 
-const (
-	DetailsAreNotLoaded = 0
-	DetailsAreLoading   = 1
-	DetailsAreLoaded    = 2
-)
-
+var areDeploymentDetailsLoaded = false
 var deploymentDetailsMap = make(map[string]string)
```

Comment:
This can stay as `var deploymentDetailsMap = map[string]string` and then you have an easy way to check if the deployment are loaded / present, which is that `deploymentDetailsMap != nil`.
That would also remove the need for that additional `areDeploymentDetailsLoaded` variable.

Reply:
Great suggestion! 👌
Now the declaration at the top looks like:
```
var deploymentDetailsMap map[string]string
```
and it's initialized in the `loadDeploymentDetails()` function:
```
func loadDeploymentDetails() {
	deploymentDetailsMap = make(map[string]string)
```
and the `getDeploymentDetailsIfLoaded` function is now:
```
func getDeploymentDetailsIfLoaded() map[string]string {
	return deploymentDetailsMap
}
```